### PR TITLE
F/refactor i18n

### PR DESCRIPTION
**Pull request by macrozone**
@rexebin  i id some small changes. what do you think?

### CODE REVIEW

File: packages/editor/src/core/components/hooks/options.ts [hunk 1]
```diff
@@ -26,22 +26,17 @@ export const OptionsContext = createContext<Options>({
...
+} => {
+  const { uiTranslator } = useOptions();
   return {
     t: (key: string) => {
-      const translation = userTranslatorFunction.t(key);
-      if (translation === undefined || translation === null) {
-        return key;
-      }
-      return translation;
+      return uiTranslator?.(key) ?? key;
```
Comment (macrozone) [inline packages/editor/src/core/components/hooks/options.ts:39]: ?? does fallback to the right operand if left ist null or undefined. So the result is the same

Comment (rexebin) [reply packages/editor/src/core/components/hooks/options.ts:39]: if the left is an empty string, it will go for the key as well?

Comment (rexebin) [reply packages/editor/src/core/components/hooks/options.ts:39]: the `??` operator only check for null or undefined I think. So I think it is the same.

Comment (macrozone) [reply packages/editor/src/core/components/hooks/options.ts:39]: yes, its the same. As you also correctly noticed that devs might indeed want to use empty strings in certain cases.

**Comment by rexebin** [comment]
@macrozone cool refactor, I am good. Learn a few tricks here. Thank you!

**Comment by rexebin** [comment]
@macrozone The `children` prop for the `Editor` component was removed from the original uiTranslator PR. It is a two-line change, do you want me to start a new PR or you would just add it during your next release? Is there a guideline for submitting adding contributor PR or you would handle it without PR? 
It was a pleasure working with you.

**Comment by macrozone** [comment]
> @macrozone The `children` prop for the `Editor` component was removed from the original uiTranslator PR. It is a two-line change, do you want me to start a new PR or you would just add it during your next release? Is there a guideline for submitting adding contributor PR or you would handle it without PR?

I'll readd it and release it

> It was a pleasure working with you.

same, thank you :-)

**Pull request merged**